doc/_gallery/plot_3_1_4_atoms_short_hamming_stft.py

```diff
@@ -21,13 +21,29 @@
 import numpy as np
 import matplotlib.pyplot as plt
 from tftb.generators import atoms
-from scipy.signal import hamming
+from scipy.signal.windows import hamming
 from tftb.processing.linear import ShortTimeFourierTransform
+import matplotlib
+matplotlib.use("TkAgg")
+
 
 coords = np.array([[45, .25, 32, 1], [85, .25, 32, 1]])
 sig = atoms(128, coords)
 x = np.real(sig)
-window = hamming(17)
-stft = ShortTimeFourierTransform(sig, n_fbins=128, fwindow=window)
-stft.run()
+
+nperseg = 17
+window = hamming(nperseg)
```

Comment:
Shouldn't the window be picked up by the STFT automatically?

Reply:
Why? There are many possible windows that can be used with the Short Time Fourier transform.
In plot 3_1_4 we use the Hamming window. In other plots we use a different window.
For use of default values see my comments further down.